### ISSUE TITLE
wrong initialization in SeparableConv2D

### ISSUE THREAD

**Issue by envytails** [keras-team/keras#9890]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ X ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ X ] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ X ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ X ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

in SeparableConv2D if I set a constant value as initializer it gives me different values ( e.g. 
weight values setting constant initializer to 100: 
[array([[[[ 0.10519278]],[[-0.09711087]],[[-0.1125772 ]]],[[[-0.18705219]],[[-0.13889125]], [[ 0.21000636]]],[[[ 0.24378067]], [[ 0.46328533]], [[-0.15319693]]]], dtype=float32), array([[[[-0.15856814]]]], dtype=float32), array([0.], dtype=float32)]

-----------------------------CODE---------------------
a2= SeparableConv2D(1, (3, 3),input_shape=[3, 3, 1], activation='linear', padding='valid', kernel_initializer=keras.initializers.Constant(100), bias_initializer=keras.initializers.Constant(0))
init = tf.global_variables_initializer()
sess.run(init)
print(a2.get_weights())

NB this behavior doesn't seem present in other layers (e.g. Conv2D)

**Comment by fchollet** [comment]
Looks like an inheritance edge case where `kernel_initializer` / `bias_initializer` args are being allowed in `SeparableConv2D` (even though the layer isn't supposed to support these arguments), doing nothing while not raising any error. This should be fixed (by raising the appropriate `TypeError`).

**Comment by envytails** [comment]
why not?
